### PR TITLE
Remove initial draft from playermodel, fix initial draft label

### LINKED ISSUE #2542: Minor bug: initial draft in gameOptions became false after end game.
Initial draft shows up correctly at the beginning of the game

image

But once the games ended, gameOptions.initialDraftVariant seems to be false.

image

image

### PR DESCRIPTION
**Pull request by vsrisuknimit**
I hope for this PR to close #2542 .

But so far I only removed initialDraft from playerModel since we already have initialdraftvariant in gameoptionsmodel.

I am still not sure what is causing the problem.

### CODE REVIEW

**Comment by vsrisuknimit** [comment]
There is something confusing between GameOptions (in Game.ts) and GameOptionsModel (in GameOptionsModel.ts).

**Comment by vsrisuknimit** [comment]
Tagging @kberg in case he might know.

**Comment by bafolts** [comment]
I believe the problem is here.

https://github.com/bafolts/terraforming-mars/blob/main/src/Game.ts#L880

For some reason we set `initialDraftVariant` to `false` here.

**Comment by vsrisuknimit** [comment]
@bafolts good find! Thank you. I will take a look to see if we need it and remove it if not.

**Comment by vsrisuknimit** [comment]
It is used here. 

https://github.com/bafolts/terraforming-mars/blob/04e9fdf5407464288a61530ffd984979a4d1a602/src/components/PlayerHome.ts#L333-L350

If we remove, the initial cards do go away when we are done with initial drafting.

image

**Comment by bafolts** [comment]
When we are done with `Phase.DRAFTING` I would set the phase to `Phase.RESEARCH`. The checks in `PlayerHome` should then use `v-if="player.phase === getInitialPhase() && player.initialDraft"`.

I would set the phase to `Phase.RESEARCH` at the top of `gotoInitialResearchPhase`.

https://github.com/bafolts/terraforming-mars/blob/main/src/Game.ts#L669

**Comment by vsrisuknimit** [comment]
Summary:
- I removed initialDraft from playerModel since we already have initialdraftvariant in gameoptionsmodel.
- I added a new Phase.InitialDraft. This is used to check whether to display the drafted cards during initial draft.
- I removed the manipulation of gameOptions.initialDraftVariant from game. Those values should be constant. It was previously used to determine whether the initial draft is done. So I rewrote that part a little.

File: src/components/PlayerHome.ts [hunk 1]
```diff
@@ -165,6 +166,12 @@ export const PlayerHome = Vue.component('player-home', {
     isEventCardShown(): boolean {
       return this.hide_event_cards !== '1';
     },
+    isInitialDraftingPhase(): boolean {
+      console.log(this.player.phase);
+      console.log(this.player.generation);
+      console.log(this.player.gameOptions.initialDraftVariant);
```
Comment (bafolts) [inline src/components/PlayerHome.ts:172]: I would remove the `console.log` as they were probably used for debugging. Is the check that it is the first generation necessary? As long as `phase === Phase.INITIALDRAFTING` that should work fine.

Comment (vsrisuknimit) [reply src/components/PlayerHome.ts:172]: My bad. I added it after the test failed.

Comment (vsrisuknimit) [reply src/components/PlayerHome.ts]: Fixed. Removed gen 1 check. I was trying to be cautious.

**Review by bafolts — APPROVED** [review]

**Pull request merged**